mongo/mongo.go

```diff
@@ -83,3 +83,40 @@ func (m *Mongo) GetRole(ctx context.Context, id string) (*models.Role, error) {
 
 	return &role, nil
 }
+
+// GetRoles retrieves all role documents from Mongo, according to the provided limit and offset.
+//Offset and limit need to  be positive or zero. Zero limit is equivalent to no limit and all items starting at the offset will be returned.
+func (m *Mongo) GetRoles(ctx context.Context, offset, limit int) (*models.Roles, error) {
+	if offset < 0 || limit < 0 {
+		return nil, apierrors.ErrLimitAndOffset
+	}
+
+	s := m.Session.Copy()
+	defer s.Close()
+	log.Event(ctx, "getting roles", log.INFO)
+
+	roles := s.DB(m.Database).C(m.Collection).Find(nil)
+	totalCount, err := roles.Count()
+	if err != nil {
+		if err == mgo.ErrNotFound {
```

Comment:
Just checking Is the error returned if there are 0 documents found?  I.e. if `err` is nil then `totalCount` will be `>= 1`?

Reply:
My understanding is that the error is not returned if there are 0 documents found.